gcp_airflow_foundations/base_class/source_template_config.py

```diff
@@ -0,0 +1,152 @@
+
+from gcp_airflow_foundations.base_class.ods_metadata_config import (
+    OdsTableMetadataConfig,
+)
+from gcp_airflow_foundations.enums.ingestion_type import IngestionType
+from gcp_airflow_foundations.enums.hds_table_type import HdsTableType
+
+from gcp_airflow_foundations.base_class.ods_table_config import OdsTableConfig
+from gcp_airflow_foundations.base_class.hds_table_config import HdsTableConfig
+from gcp_airflow_foundations.base_class.facebook_table_config import FacebookTableConfig
+from gcp_airflow_foundations.base_class.template_ingestion_options_config import TemplateIngestionOptionsConfig
+from gcp_airflow_foundations.base_class.column_udf_config import ColumnUDFConfig
+
+
+@dataclass
+class SourceTemplateConfig:
+    """
+    Template configuration data class.
+
+    Attributes:
+        table_names : List of table name. Used for Dag Id/s.
+        dest_table_override: Optional target table name. If None, use table_name instead.
+        surrogate_keys : Keys used to identify unique records when merging into ODS.
+        column_mapping : Mapping used to rename columns.
+        cluster_fields: The fields used for clustering. BigQuery supports clustering for both partitioned and non-partitioned tables.
+        landing_zone_table_name_override_template: Optional staging zone table name template.
+        column_casting : Mapping used to cast columns into a specific data type. Note column name uses that of the landing zone table.
+        ods_config : ODS table configuration. See :class:`gcp_airflow_foundations.base_class.ods_table_config.OdsTableConfig`.
+        hds_config : HDS table configuration. See :class:`gcp_airflow_foundations.base_class.hds_table_config.HdsTableConfig`.
+        template_ingestion_options: Configuration for template-level ingestion.
+        facebook_table_config: Extra options for ingesting data from the Facebook API.
+        extra_options: Field for storing additional configuration options.
+        start_date : Start date override for DAG
+        start_date_tz : Timezone
+        version : The Dag version for the table. Can be incremented if logic changes.
+        catchup : Passed to a dag [see doc](https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#catchup).
+            Defaults to True. May want to change it to False if Dag version is changed, and we don't want to rerun past dags.
+    """
+
+    ingestion_type: IngestionType  # FULL or INCREMENTAL
+    dest_table_override: Optional[str]
+    surrogate_keys: List[str]
+    column_mapping: Optional[dict]
+    cluster_fields: Optional[List[str]]
+    landing_zone_table_name_override_template: Optional[str]
+    column_casting: Optional[dict]
+    new_column_udfs: Optional[dict]
+    hds_config: Optional[HdsTableConfig]
+    facebook_table_config: Optional[FacebookTableConfig]
+    start_date: Optional[str]
+    extra_options: dict = field(default_factory=dict)
+    table_names: list = field(default_factory=list)
```

Comment:
Isn't table_name suppose to be part of TemplateIngestionOptionsConfig? Seems to have same purpose

Reply:
can throw it in there if it is cleaner - the SourceTableConfig had table_name on the top level, so figured would keep it semi-consistent